index.js

```diff
@@ -0,0 +1,387 @@
+        const key = document.createElement('button');
+        key.className = `key ${keyboardKeys[i][j][0]}`;
+        row.append(key);
+
+        const spanEn = document.createElement('span');
+        const spanEnUp = document.createElement('span');
+        const spanEnDown = document.createElement('span');
+        const spanLt = document.createElement('span');
+        const spanLtUp = document.createElement('span');
+        const spanLtDown = document.createElement('span');
+
+        let [langOn, langOff] = [' on', ' off'];
+        if (localStorage.getItem('virtualKeyboardLang') === 'EN') {
+          langOn = ' on';
+          langOff = ' off';
+        } else {
+          langOn = ' off';
+          langOff = ' on';
+        }
+        spanEn.className = keyboardKeys[i][j][1] + langOn;
+        spanLt.className = keyboardKeys[i][j][1] + langOff;
+
+        key.append(spanEn);
+        key.append(spanLt);
+
+        spanLtDown.className = 'case-shown';
+        spanLt.append(spanLtDown);
+        spanLtDown.insertAdjacentText('afterbegin', keyboardKeys[i][j][2]);
+
+        spanLtUp.className = 'case-hidden';
+        spanLt.append(spanLtUp);
+        spanLtUp.insertAdjacentText('afterbegin', keyboardKeys[i][j][3]);
+
+        spanEnDown.className = 'case-shown';
+        spanEn.append(spanEnDown);
+        spanEnDown.insertAdjacentText('afterbegin', keyboardKeys[i][j][4]);
+
+        spanEnUp.className = 'case-hidden';
+        spanEn.append(spanEnUp);
+        spanEnUp.insertAdjacentText('afterbegin', keyboardKeys[i][j][5]);
+      }
+    }
+  }
+
+  caseUp() {
+    this.isShiftPress = true;
+    document.querySelectorAll('.on').forEach(key => {
+      key.children[0].classList.remove('case-shown');
+      key.children[0].classList.add('case-hidden');
+      key.children[1].classList.add('case-shown');
+      key.children[1].classList.remove('case-hidden');
+    });
+  }
+
+  caseDown() {
+    this.isShiftPress = false;
+    document.querySelectorAll('.on').forEach(key => {
+      key.children[0].classList.add('case-shown');
+      key.children[0].classList.remove('case-hidden');
+      key.children[1].classList.remove('case-shown');
+      key.children[1].classList.add('case-hidden');
+    });
+  }
+
+  setCaretPosition(pos) {
+    if (this.textArea.setSelectionRange) {
+      this.textArea.focus();
+      this.textArea.setSelectionRange(pos, pos);
+    } else if (this.textArea.createTextRange) {
+      const range = this.textArea.createTextRange();
+      range.collapse(true);
+      range.moveEnd('', pos);
+      range.moveStart('', pos);
+      range.select();
+    }
+  }
+
+  activeBtnHighlighting(specialBtnEl) {
+    if (this.isShiftPress) {
+      this.isShiftPress = true;
+      specialBtnEl.classList.add('active');
+    } else {
+      this.isShiftPress = false;
+      specialBtnEl.classList.remove('active');
+    }
+  }
+
+  pageLangChanging() {
+    if (localStorage.getItem('virtualKeyboardLang') === 'EN') {

```

Comment:
you call to get value from local storage multiple times, I suggest to just declare a const and call it once, and reuse the constant rather than calling storage multiple times.

`const localStorageValue = localStorage.getItem('virtualKeyboardLang') `